service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/StandardDirectionsOrderController.java

```diff
@@ -187,119 +227,127 @@ public CallbackResponse handleUploadMidEvent(@RequestBody CallbackRequest reques
         CaseData caseData = getCaseData(caseDetails);
         CaseData caseDataBefore = getCaseDataBefore(request);
 
-        StandardDirectionOrder order = sdoService.buildTemporarySDO(
-            caseData,
-            caseDataBefore.getStandardDirectionOrder()
-        );
+        StandardDirectionOrder sdo = sdoService.buildTemporarySDO(caseData, caseDataBefore.getStandardDirectionOrder());
 
-        caseDetails.getData().put(STANDARD_DIRECTION_ORDER_KEY, order);
+        caseDetails.getData().put(STANDARD_DIRECTION_ORDER_KEY, sdo);
+        // work around as I could not get the page hiding when using the [STATE] metadata field
+        caseDetails.getData().put(
+            "showNoticeOfProceedings", YesNo.from(GATEKEEPING == caseData.getState())
```

Comment:
as mentioned could look into re-using existing 'state' field instead of new case field

Reply:
see previous